StripeCore/StripeCore/Source/Helpers/URLSession+Async.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  URLSession+Async.swift
+//  StripeCore
+//
+//  Created by Yuki Tokuhiro on 1/6/23.
+//
+
+import Foundation
+
+@available(iOS, deprecated: 15.0, message: "Use the built-in API instead")
+extension URLSession {
+    /// An Xcode 13 compatible version of `data(for:)`
+    /// Taken from https://www.swiftbysundell.com/articles/making-async-system-apis-backward-compatible/
+    @_spi(STP) public func stp_data(for request: URLRequest) async throws -> (Data, URLResponse) {
```

Comment:
should we use an `if #available` check and use the new api if it's available? I'd prefer to only fallback to the wrapper when necessary.